preferences.go

```diff
@@ -0,0 +1,31 @@
+package gapi
+
+// NavLink represents a Grafana nav link.
+type NavLink struct {
+	ID     string `json:"id,omitempty"`
+	Text   string `json:"text,omitempty"`
+	URL    string `json:"url,omitempty"`
+	Target string `json:"target,omitempty"`
+}
+
+// NavbarPreference represents a Grafana navbar preference.
+type NavbarPreference struct {
+	SavedItems []NavLink `json:"savedItems"`
+}
+
+// QueryHistoryPreference represents a Grafana query history preference.
+type QueryHistoryPreference struct {
+	HomeTab string `json:"homeTab"`
+}
+
+// Preferences represents Grafana preferences.
+type Preferences struct {
```

Comment:
Previously, an incomplete and mistaken `type Preferences` implementation lived in `team.go`. I've...

* moved it to a shared, common location in `preferences.go`
* added the additional fields that may be present
* corrected the `HomeDashboardID` field to utilize a `homeDashboardId` field (rather than a `homeDashboardID` field, which [appears to have been incorrect](https://github.com/grafana/grafana/blob/main/pkg/api/dtos/prefs.go#L9))

Reply:
What I understand from this is that the previous version wasn't properly passing the `HomeDashboardID` value to Grafana, then?

If that's the case then this is great. If not, then it means something might break. Can you confirm if it was working before or not?

Reply:
@inkel I don't believe anything will break. Based on my integration testing (see PR #105), I believe both `json:homeDashboardID` and `json:homeDashboardId` will work correctly.

However, I believe the Grafana API itself uses the field `homeDashboardId` (and not `homeDashboardID`), as evidenced by the following, where the `ff0c251f0c51` Grafana container resulted from the PR #105 integration testing technique`:

```
$ docker ps
CONTAINER ID   IMAGE                      COMMAND         CREATED          STATUS          PORTS                    NAMES
ff0c251f0c51   grafana/grafana:9.0.2      "/run.sh"       42 minutes ago   Up 42 minutes   3000/tcp                 grafana-api-golang-client_grafana_1

$ docker exec -u 0 --interactive --tty ff0c251f0c51 /bin/bash
bash-5.1# apk add curl
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.15/community/x86_64/APKINDEX.tar.gz
(1/4) Installing brotli-libs (1.0.9-r5)
(2/4) Installing nghttp2-libs (1.46.0-r0)
(3/4) Installing libcurl (7.80.0-r2)
(4/4) Installing curl (7.80.0-r2)
Executing busybox-1.34.1-r5.trigger
Executing glibc-bin-2.30-r0.trigger
/usr/glibc-compat/sbin/ldconfig: /usr/glibc-compat/lib/ld-linux-x86-64.so.2 is not a symbolic link

OK: 28 MiB in 38 packages

bash-5.1# curl -u admin:admin http://localhost:3000/api/teams/search?query=
{"totalCount":1,"teams":[{"id":8,"orgId":1,"name":"foo","email":"foo@bar.com","avatarUrl":"/avatar/f3ada405ce890b6f8204094deb12d8a8","memberCount":1,"permission":0,"accessControl":null}],"page":1,"perPage":1000}

bash-5.1# curl -u admin:admin htt/teams/8/preferences
{"theme":"","homeDashboardId":123,"timezone":"","weekStart":"","navbar":{"savedItems":null},"queryHistory":{"homeTab":""}}
```